### PR TITLE
Modifed Druva_FindDevice_Command in druva.py,Druva Ransomware Respons…

### PR DESCRIPTION
**Pull request by pk-druva**
…e playbook and deleted Druva Ransomware Response-user-resource-url playbook

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by pk-druva** [comment]
> @pk-druva - Which one is the right PR? This one or #25556?

this one

**Comment by daryakoval** [comment]
Hi, Thanks for the contribution! 
I saw a lot of "strange" changes in the PR in the python file: Packs/Druva/Integrations/Druva/Druva.py. 
Like formatting: 
image

Please remove all unnecessary changes from the python file, so it will be easier to review it.

File: Packs/Druva/Playbooks/Druva-Ransomware-Response-User-resource-URL.yml [hunk 1]
```diff
@@ -1,546 +0,0 @@
-id: Druva-Ransomware-Response-User-resource-URL
-version: -1
```
Comment (daryakoval) [inline Packs/Druva/Playbooks/Druva-Ransomware-Response-User-resource-URL.yml:2]: Please do not delete existing content items. You can deprecate it instead: https://xsoar.pan.dev/docs/reference/articles/deprecation-process-and-hidden-packs#how-to-deprecate-a-playbook

File: Packs/Druva/Playbooks/Druva-Ransomware-Response-User-resource-URL_README.md [hunk 2]
```diff
@@ -1,42 +0,0 @@
-Automate response actions like quarantining effected resources or snapshots to stop the spread of ransomware and avoid reinfection or contamination spread.
-
```
Comment (daryakoval) [inline Packs/Druva/Playbooks/Druva-Ransomware-Response-User-resource-URL_README.md:2]: Please do not delete this file.

File: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml [hunk 3]
```diff
@@ -1,491 +1,667 @@
-id: Druva-Ransomware-Response
-version: -1
+id: 083d0b4d-3f94-4214-8ee0-6ed616f3aec5
+version: 10
+vcShouldKeepItemLegacyProdMachine: false
```
Comment (daryakoval) [inline Packs/Druva/Playbooks/Druva-Ransomware-Response.yml:3]:
```suggestion
id: Druva-Ransomware-Response
version: -1
```

File: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml [hunk 4]
```diff
@@ -1,491 +1,667 @@
...
-- contextPath: Druva.viewedQuarantineRange
-  description: Druva Range Output
-  type: unknown
-tests:
-- Druva-Test
-fromversion: 6.0.0
+  description: 'Date is used to quarantine device '
+  playbookInputQuery: null
+outputs: []
+sourceplaybookid: 97d62768-a09d-4f5f-83d1-8d07ddec12f4
```
Comment (daryakoval) [inline Packs/Druva/Playbooks/Druva-Ransomware-Response.yml:667]:
```suggestion
```

File: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml [hunk 5]
```diff
@@ -1,491 +1,667 @@
...
   required: false
-  description: FIle Hash of ransomware Detected
-  playbookInputQuery:
-outputs:
-- contextPath: Druva.viewedQuarantineRange
-  description: Druva Range Output
-  type: unknown
-tests:
-- Druva-Test
-fromversion: 6.0.0
```
Comment (daryakoval) [inline Packs/Druva/Playbooks/Druva-Ransomware-Response.yml:491]: Please add back the `fromversion: 6.0.0` key.

File: Packs/Druva/Integrations/Druva/Druva.py [hunk 6]
```diff
@@ -34,9 +34,14 @@ def get_quarantineRanges(self):
         return self._http_request(method='GET', url_suffix="/realize/ransomwarerecovery/v1/quarantineranges",
                                   resp_type='response')
 
-    def get_findDevice(self, search_string):
+    def get_findFsNasDevice(self, search_string):
         params = {'hostname': search_string}
-        return self._http_request(method='GET', url_suffix="/realize/ransomwarerecovery/v1/search/backupset",
+        return self._http_request(method='GET', url_suffix=f"/realize/ransomwarerecovery/v1/search/backupset",
```
Comment (daryakoval) [inline Packs/Druva/Integrations/Druva/Druva.py:39]:
```suggestion
        return self._http_request(method='GET', url_suffix="/realize/ransomwarerecovery/v1/search/backupset",
```

File: Packs/Druva/Integrations/Druva/Druva.py [hunk 7]
```diff
@@ -34,9 +34,14 @@ def get_quarantineRanges(self):
...
+                                  params=params, resp_type='response')
+
+    def get_findVMDevice(self, search_string):
+        params = {'hostname': search_string,'serverTypes[]':3}
+        return self._http_request(method='GET', url_suffix=f"/realize/ransomwarerecovery/v1/search/backupset",
```
Comment (daryakoval) [inline Packs/Druva/Integrations/Druva/Druva.py:44]:
```suggestion
        return self._http_request(method='GET', url_suffix="/realize/ransomwarerecovery/v1/search/backupset",
```

File: Packs/Druva/Integrations/Druva/Druva.py [hunk 8]
```diff
@@ -133,14 +138,17 @@ def Druva_ListQuarantineRanges_Command(clientObj):
...
+    responseVM = clientObj.get_findVMDevice(search_string)
     statusCode = response.status_code
     if (statusCode == 200):
         responseJson = response.json()
-        results = CommandResults(
-            readable_output=tableToMarkdown('Found Druva Devices', responseJson['resources']),
-            outputs={"Druva.Resource(val.resourceID == obj.resourceID)": responseJson['resources']},
-            raw_response=responseJson
+        responseVMJson = responseVM.json()
+        finalResponce=responseJson['resources']+responseVMJson['resources']
```
Comment (daryakoval) [inline Packs/Druva/Integrations/Druva/Druva.py:147]:
```suggestion
        finalResponse=responseJson.get('resources', []) + responseVMJson.get('resources', [])
```

File: Packs/Druva/Integrations/Druva/Druva.py [hunk 9]
```diff
@@ -133,14 +138,17 @@ def Druva_ListQuarantineRanges_Command(clientObj):
...
+        results=CommandResults(
+            readable_output = tableToMarkdown('Found Druva Devices', finalResponce),
+            outputs = {"Druva.Resource(val.resourceID == obj.resourceID)": finalResponce},
+            raw_response = responseJson|responseVMJson
```
Comment (daryakoval) [inline Packs/Druva/Integrations/Druva/Druva.py:151]:
```suggestion
            readable_output = tableToMarkdown('Found Druva Devices', finalResponse),
            outputs = {"Druva.Resource(val.resourceID == obj.resourceID)": finalResponse},
            raw_response = responseJson | responseVMJson
```

**Review by daryakoval — CHANGES_REQUESTED** [review]
Hi, Thanks for your fast response. 
Please see my comments. 
In addition, please generate Release Notes using: `demisto-sdk update-release-notes` command.

File: Packs/Druva/Playbooks/Druva-Ransomware-Response-User-resource-URL.yml (hunk 1 above)
Comment (pk-druva) [reply Packs/Druva/Playbooks/Druva-Ransomware-Response-User-resource-URL.yml:2]: This was never meant for publishing.
Hence, we need to delete this.

File: Packs/Druva/Playbooks/Druva-Ransomware-Response-User-resource-URL_README.md (hunk 2 above)
Comment (pk-druva) [reply Packs/Druva/Playbooks/Druva-Ransomware-Response-User-resource-URL_README.md:2]: This was never meant for publishing.
Hence, we need to delete this.

File: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml (hunk 4 above)
Comment (pk-druva) [reply Packs/Druva/Playbooks/Druva-Ransomware-Response.yml]: Change added

File: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml (hunk 5 above)
Comment (pk-druva) [reply Packs/Druva/Playbooks/Druva-Ransomware-Response.yml:491]: Change added

File: Packs/Druva/Integrations/Druva/Druva.py (hunk 6 above)
Comment (pk-druva) [reply Packs/Druva/Integrations/Druva/Druva.py]: Change added

File: Packs/Druva/Integrations/Druva/Druva.py (hunk 8 above)
Comment (pk-druva) [reply Packs/Druva/Integrations/Druva/Druva.py]: Change added

File: Packs/Druva/Integrations/Druva/Druva.py (hunk 9 above)
Comment (pk-druva) [reply Packs/Druva/Integrations/Druva/Druva.py]: Change added

**Review by efelmandar — COMMENTED** [review]
Hi @pk-druva thanks for your contribution! We appreciate your efforts in creating this content. However, as Darya mentioned, deleting content is strongly discouraged, as there might be existing users who may already be using this content. Additionally, it appears that the new playbook you have added has "deprecated: True" in the YAML file, which doesn't make sense if you want this playbook to be used.
Please review these issues and make any necessary changes or let me know if you think otherwise so we can discuss it to find the right solution here.

**Comment by pk-druva** [comment]
> Hi @pk-druva thanks for your contribution! We appreciate your efforts in creating this content. However, as Darya mentioned, deleting content is strongly discouraged, as there might be existing users who may already be using this content. Additionally, it appears that the new playbook you have added has `deprecated: True` in the YAML file, which doesn't make sense if you want this playbook to be used. Please review these issues and make any necessary changes or let me know if you think otherwise so we can discuss it to find the right solution here.

Sorry for that i have removed deprecated: True from file

File: Packs/Druva/ReleaseNotes/1_1_3.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
+
+#### Integrations
+
+##### Druva Ransomware Response
+
+- Fixed command to search VMware resources
```
Comment (daryakoval) [inline Packs/Druva/ReleaseNotes/1_1_3.md:6]:
```suggestion
- Fixed the command ***druva-find-device*** to search VMware resources.
- Updated the Docker image to: *demisto/python3:3.10.10.51930*.
```

**Comment by daryakoval** [comment]
Please fix the following validations: 

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '0'-> 'task' Please add the field "description" to the path:  'tasks'-> '0'-> 'task' in the yml.

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '1'-> 'task' Please add the field "description" to the path:  'tasks'-> '1'-> 'task' in the yml.

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '2'-> 'task' Please add the field "description" to the path:  'tasks'-> '2'-> 'task' in the yml.

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '3'-> 'task' Please add the field "description" to the path:  'tasks'-> '3'-> 'task' in the yml.

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '9'-> 'task' Please add the field "description" to the path:  'tasks'-> '9'-> 'task' in the yml.

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '13'-> 'task'
Please add the field "description" to the path:  'tasks'-> '13'-> 'task' in the yml.

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '15'-> 'task'
Please add the field "description" to the path:  'tasks'-> '15'-> 'task' in the yml.

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '16'-> 'task'
Please add the field "description" to the path:  'tasks'-> '16'-> 'task' in the yml.

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [ST107] - Missing the field "description" in Path: 'tasks'-> '17'-> 'task'
Please add the field "description" to the path:  'tasks'-> '17'-> 'task' in the yml.

- [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response_README.md: [RM108] - Error in readme image: 
The following image relative path is not valid, please recheck it:
Druva-Ransomware-Response.

- Please update the docker image tag to - 3.10.10.51930. 


- Please fix the Flake8 errors:
###############
 Flake8 errors 
###############

Druva
Druva.py:43:44: E231 missing whitespace after ','
Druva.py:43:60: E231 missing whitespace after ':'
Druva.py:147:22: E225 missing whitespace around operator
Druva.py:148:16: E225 missing whitespace around operator
Druva.py:149:28: E251 unexpected spaces around keyword / parameter equals
Druva.py:149:30: E251 unexpected spaces around keyword / parameter equals
Druva.py:150:20: E251 unexpected spaces around keyword / parameter equals
Druva.py:150:22: E251 unexpected spaces around keyword / parameter equals
Druva.py:151:25: E251 unexpected spaces around keyword / parameter equals
Druva.py:151:27: E251 unexpected spaces around keyword / parameter equals
Druva.py:502:11: W292 no newline at end of file
Druva_test.py:29:6: E231 missing whitespace after ','

**Comment by daryakoval** [comment]
Please handle this one as well. 
[ERROR]: Packs/Druva/ReleaseNotes/1_1_3.md: [RN107] - No release note entry was found for the playbook "Druva-Ransomware-Response-User-resource-URL" in the Druva pack. Please rerun the update-release-notes command without -u to generate an updated template. If you are trying to exclude an item from the release notes, please refer to the documentation found here - https://xsoar.pan.dev/docs/integrations/changelog#excluding-items

**Comment by daryakoval** [comment]
Hi, 
We got 2 new validation errors: 
[ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [PB118] - Playbook Druva-Ransomware-Response contains inputs that are not used in any of its tasks: DateOfOccurrence

[ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [PB119] - Playbook Druva-Ransomware-Response uses inputs that do not appear in the inputs section: DateOfOccurrence}, ResourceName}, SiteURL}, UserName}

**Comment by pk-druva** [comment]
> Hi, We got 2 new validation errors: [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [PB118] - Playbook Druva-Ransomware-Response contains inputs that are not used in any of its tasks: DateOfOccurrence
> 
> [ERROR]: Packs/Druva/Playbooks/Druva-Ransomware-Response.yml: [PB119] - Playbook Druva-Ransomware-Response uses inputs that do not appear in the inputs section: DateOfOccurrence}, ResourceName}, SiteURL}, UserName}

DateOfOccurrence is getting used in one of the tasks 
image


And about second error can you please elaborate a little on how to fix this

**Review by daryakoval — APPROVED** [review]

**Review by daryakoval — APPROVED** [review]

**Pull request merged**